src/app/Dashboard/Topic.js

```diff
@@ -33,24 +43,52 @@ export class Topic extends React.Component {
     const { title, topicId } = this.props;
 
     return (
-      <div
-        className="topic-wrap"
-        onMouseEnter={this.toggleHidden}
-        onMouseLeave={this.toggleHidden}
-      >
-        <button
-          className="topic-btn"
-          onClick={() => console.log('click through to topic/:id', topicId)}
+      <>
+        <div
+          className="topic-wrap"
+          onMouseEnter={this.toggleHidden}
+          onMouseLeave={this.toggleHidden}
         >
-          {title}
-        </button>
-        {!this.state.isHidden && (
-          <div className="edit-delete-topic-options">
-            <button onClick={this.editTopic}>Edit</button>
-            <button onClick={this.deleteTopic}>Delete</button>
-          </div>
-        )}
-      </div>
+          {this.state.editing ? (
+            <>
+              <form
+                className="edit-topic-form"
+                onSubmit={e => this.submitEdit(e)}
+              >
+                <label>Topic Name</label>
+                <input
+                  type="text"
+                  name="topicTitle"
+                  defaultValue={this.props.title}
+                />
+              </form>
+              {!this.state.isHidden && (
```

Comment:
Something I've learned: double negatives like this line (not-not-shown) are hard for us mortals to parse.

For example, you could call the state object something like `showForm` so the code would read like:

```js
this.state.showForm && {...}
```

Totally optional. Just a thought.